### ISSUE TITLE
Bypass proxy for local URLs

### ISSUE THREAD

**Issue by thedoginthewok** [larshp/abapGit#3219]
I'd like a feature where you can add exemptions for the proxy settings.

Example from macOS proxy settings:
<img width="582" alt="Bildschirmfoto 2020-02-21 um 21 38 50" src="https://user-images.githubusercontent.com/6908247/75070107-dcb6ae80-54f2-11ea-9208-ac57c0007124.png">

My suggestion would be adding a select option directly under the proxy settings in abapGit, so that wildcards and multiple entries are supported in native SAP style:
![abapgit_proxy](https://user-images.githubusercontent.com/6908247/75070254-27d0c180-54f3-11ea-88c6-d0285bfa13bc.png)

I've already implemented the user exit and it's working well, but I think this makes sense as a standard feature.

```abap
CLASS zcl_abapgit_user_exit DEFINITION PUBLIC FINAL CREATE PUBLIC.
  PUBLIC SECTION.
    INTERFACES:
      zif_abapgit_exit.
  PROTECTED SECTION.
  PRIVATE SECTION.
    DATA:
      mr_proxy_bypass_hosts TYPE RANGE OF string.
    METHODS:
      is_proxy_necessary IMPORTING iv_repo_url TYPE string
                         RETURNING VALUE(rv)   TYPE abap_bool.
ENDCLASS.



CLASS zcl_abapgit_user_exit IMPLEMENTATION.

  METHOD is_proxy_necessary.
* ---------------------------------------------------------------------
    IF mr_proxy_bypass_hosts IS INITIAL.
      SELECT sign, opti AS option, low, high
      FROM tvarvc
      WHERE name = 'ZABAPGIT_BYPASS_PROXY'
      AND   type = 'S'
      INTO CORRESPONDING FIELDS OF TABLE @mr_proxy_bypass_hosts.
    ENDIF.
    IF iv_repo_url NOT IN mr_proxy_bypass_hosts.
      rv = abap_true.
    ENDIF.

* ---------------------------------------------------------------------
  ENDMETHOD.


  METHOD zif_abapgit_exit~change_proxy_port.
* ---------------------------------------------------------------------
    IF NOT is_proxy_necessary( iv_repo_url ).
      CLEAR cv_proxy_port.
    ENDIF.

* ---------------------------------------------------------------------
  ENDMETHOD.


  METHOD zif_abapgit_exit~change_proxy_url.
* ---------------------------------------------------------------------
    IF NOT is_proxy_necessary( iv_repo_url ).
      CLEAR cv_proxy_url.
    ENDIF.

* ---------------------------------------------------------------------
  ENDMETHOD.

ENDCLASS.
```

**Comment by jfilak** [comment]
Is the table `tvarvc` available everywhere?

**Comment by thedoginthewok** [comment]
It is:
![grafik](https://user-images.githubusercontent.com/6908247/75082964-7f345900-5516-11ea-9b10-37ef74c52589.png)

But it should be included in the abapGit settings xml in db table ZABAPGIT.

**Comment by larshp** [comment]
there is a lot of framework in abapGit to store the values, suggest using that

**Comment by thedoginthewok** [comment]
> 
> 
> there is a lot of framework in abapGit to store the values, suggest using that

I agree

**Comment by thedoginthewok** [comment]
I've forked the abapGit repo and created a [Branch](https://github.com/thedoginthewok/abapGit/tree/proxy_bypass) and made some changes. It works on my local system, but the sapevent in the settings page of my added button does not work on my companies citrix machine.

Any idea why it isn't working? My PC is running Win10 with IE11 and the citrix terminal server is on Win Server 2008 with IE11.

This is how I've added the button:
```abap
ro_html->add( |<a href="sapevent:{ c_action-change_proxy_bypass }">| ).
ro_html->add( |<button name="proxy_bypass" type="button" class="grey-set">Maintain</button>| ).
ro_html->add( |</a>| ).
```
